src/main/java/org/alfresco/repo/rendition2/RenditionService2Impl.java

```diff
@@ -374,41 +382,115 @@ public void failure(NodeRef sourceNodeRef, RenditionDefinition2 renditionDefinit
+        {
+            TransformDefinition transformDefinition = (TransformDefinition)renditionDefinition;
+            String targetMimetype = transformDefinition.getTargetMimetype();
+            if (AsynchronousExtractor.isMetadataExtractMimetype(targetMimetype))
+            {
+                consumeExtractedMetadata(sourceNodeRef, sourceContentHashCode, transformInputStream, transformDefinition, transformContentHashCode);
+            }
+            else if (AsynchronousExtractor.isMetadataEmbedMimetype(targetMimetype))
+            {
+                consumeEmbeddedMetadata(sourceNodeRef, sourceContentHashCode, transformInputStream, transformDefinition, transformContentHashCode);
+            }
+            else
+            {
+                consumeTransformReply(sourceNodeRef, transformInputStream, transformDefinition, transformContentHashCode);
+            }
+        }
+        else
+        {
+            consumeRendition(sourceNodeRef, sourceContentHashCode, transformInputStream, renditionDefinition, transformContentHashCode);
+        }
+    }
+
+    private void consumeExtractedMetadata(NodeRef nodeRef, int sourceContentHashCode, InputStream transformInputStream,
+                                          TransformDefinition transformDefinition, int transformContentHashCode)
+    {
+        if (transformInputStream == null)
         {
             if (logger.isDebugEnabled())
             {
-                TransformDefinition transformDefinition = (TransformDefinition)renditionDefinition;
-                String transformName = transformDefinition.getTransformName();
-                String replyQueue = transformDefinition.getReplyQueue();
-                String clientData = transformDefinition.getClientData();
-                boolean success = transformInputStream != null;
-                logger.info("Reply to " + replyQueue + " that the transform " + transformName +
-                        " with the client data " + clientData + " " + (success ? "was successful" : "failed."));
+                logger.debug("Ignore transform for metadata extraction on " + nodeRef + " as it failed");
+            }
+        }
+        else if (transformContentHashCode != sourceContentHashCode)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Ignore transform for metadata extraction on " + nodeRef + " as it is no longer needed");
+            }
+        }
+        else
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Set the metadata extraction on " + nodeRef);
+            }
+            asynchronousExtractor.setMetadata(nodeRef, transformInputStream);
+        }
+    }
+
+    private void consumeEmbeddedMetadata(NodeRef nodeRef, int sourceContentHashCode, InputStream transformInputStream,
+                                         TransformDefinition transformDefinition, int transformContentHashCode)
```

Comment:
Unused parameter transformDefinition

Reply:
Correct. It matches consumeTransformReply. This is one of the things I hate about the repo. We cannot refactor later, and as a result we end up putting stuff in just in case it is needed.